File: src/Starter.Tests/TestAppBuilder.cs
using Avalonia;
using Avalonia.Headless;

[assembly: AvaloniaTestApplication(typeof(Starter.Tests.TestAppBuilder))]
[assembly: AvaloniaTestIsolation(AvaloniaTestIsolationLevel.PerAssembly)]

namespace Starter.Tests;

public class TestAppBuilder
{
    public static AppBuilder BuildAvaloniaApp() =>
        AppBuilder
            .Configure(() => new App { IsTestMode = true })
            .UseHeadless(new AvaloniaHeadlessPlatformOptions());
}

File: src/Starter.Tests/TestAppBuilder.fs
module Starter.Tests.Program

open Avalonia
open Avalonia.Headless
open NUnit.Framework

type TestAppBuilder() =
    static member BuildAvaloniaApp() =
        AppBuilder
            .Configure(fun () -> Starter.App(IsTestMode = true))
            .UseHeadless(AvaloniaHeadlessPlatformOptions())

[<assembly: AvaloniaTestApplication(typeof<TestAppBuilder>)>] ()

[<SetUp>]
let setup () = ()

[<EntryPoint>]
let main _ = 0
